Utility/UtilityFunctions.cpp

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+/ UtilityFunctions.cpp
+/
+/ Copyright (c) 2019
+/
+/ Permission is hereby granted, free of charge, to any person obtaining a copy
+/ of this software and associated documentation files (the "Software"), to deal
+/ in the Software without restriction, including without limitation the rights to
+/ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+/ of the Software, and to permit persons to whom the Software is furnished to
+/ do so, subject to the following conditions:
+/
+/ The above copyright notice and this permission notice shall be included in all
+/ copies or substantial portions of the Software.
+/
+/ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+/ EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+/ OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+/ NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+/ HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+/ WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+/ FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+/ OTHER DEALINGS IN THE SOFTWARE.
+/
+******************************************************************************/
+
+#include "stdafx.h"
+#include "UtilityFunctions.hpp"
+
+// wraps GetEnvelopeStateChunk() to make it more save
+// throws _bad_get_env_chunk_big on failure
+// until maybe it gets enhanced one day
+// https://forum.cockos.com/showthread.php?p=2142245#post2142245
+std::string _GetEnvelopeStateChunkBig(TrackEnvelope* envelope, bool isUndo /*= false*/)
+{
+	_bad_get_env_chunk_big exExceeding100MiB("_GetEnvelopeStateChunkBig() exception (>  100 MiB)");
+	_bad_get_env_chunk_big exBad_alloc("_GetEnvelopeStateChunkBig() exception (bad_alloc)");
```

Comment:
There is no need to construct the exception object ahead of time. `throw bad_get_env_chunk_big("Reason")` should be enough.